contracts/rewards/emission_manager_library.cairo

```diff
@@ -0,0 +1,214 @@
+        user : felt, claimer : felt
+    ):
+        Ownable.assert_only_owner()
+
+        let (reward_controller_address) = EmissionManager_rewards_controller.read()
+
+        # TODO
+        # Properly implement set_claimer in reward
+        # IRewardsController.set_claimer(
+        #     contract_address=reward_controller_address, user=user, claimer=claimer
+        # )
+        return ()
+    end
+
+    func set_rewards_controller{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        rewards_controller : felt
+    ):
+        Ownable.assert_only_owner()
+
+        EmissionManager_rewards_controller.write(rewards_controller)
+
+        return ()
+    end
+
+    func set_emission_admin{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        reward : felt, admin : felt
+    ):
+        Ownable.assert_only_owner()
+
+        let (old_admin) = EmissionManager_emission_admins.read(reward)
+        EmissionManager_emission_admins.write(reward, admin)
+
+        emission_admin_updated.emit(old_admin=old_admin, new_admin=admin)
+
+        return ()
+    end
+
+    # Getters
+
+    func get_rewards_controller{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        ) -> (res : felt):
+        let (res) = EmissionManager_rewards_controller.read()
+        return (res)
+    end
+
+    func get_emission_admin{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        reward : felt
+    ) -> (res : felt):
+        let (res) = EmissionManager_emission_admins.read(reward)
+        return (res)
+    end
+
+    # Internals
+
+    func _get_rewards_from_config{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(
+        config_len : felt,
+        config : RewardsDataTypes.RewardsConfigInput*,
+        rewards_len : felt,
+        rewards : felt*,
+    ) -> (
+        config_len : felt,
+        config : RewardsDataTypes.RewardsConfigInput*,
+        rewards_len : felt,
+        rewards : felt*,
+    ):
+        if config_len == 0:
+            return (0, &config[0], rewards_len, rewards)
+        end
+
+        let config_value = config[0]
+
+        memcpy(&rewards[rewards_len], &config_value.reward_address, 1)
+
+        return _get_rewards_from_config(
+            config_len - 1, &config[rewards_len + 1], rewards_len + 1, rewards
+        )
+    end
+
+    func _validate_emission_admins{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(rewards_len : felt, rewards : felt*):
+        stream.foreach(_validate_emission_admin_wrapper, rewards_len, rewards)
+        return ()
+    end
+
+    func _validate_emission_admin_wrapper{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(index : felt, el : felt*):
+        let reward = [el]
+        with_attr error_message("Sender is not emission admin of pool: {reward}."):
+            assert_only_emission_admin(reward)
+        end
```

Comment:
couldn't we just replace these three lines with `assert_only_emission_admin(reward)`?

Reply:
Done, and modified assert_only_emission_admin to print the reward.